spec/lita/services/market_manager_spec.rb

```diff
@@ -0,0 +1,192 @@
+require 'spec_helper'
+require 'pry'
+require 'dotenv/load'
+
+describe Lita::Services::MarketManager, lita: true do
+  let(:robot) { Lita::Robot.new(registry) }
+  let(:karmanager) do
+    Lita::Services::Karmanager.new(Lita::Handlers::LunchReminder.new(robot).redis)
+  end
+  let(:lunch_assigner) do
+    Lita::Services::LunchAssigner.new(Lita::Handlers::LunchReminder.new(robot).redis, karmanager)
+  end
+  let(:subject) do
+    described_class.new(
+      Lita::Handlers::LunchReminder.new(robot).redis,
+      lunch_assigner,
+      karmanager
+    )
+  end
+  let(:fdom) { Lita::User.create(127, mention_name: 'fdom') }
+  let(:andres) { Lita::User.create(137, mention_name: 'andres') }
+  let(:oscar) { Lita::User.create(147, mention_name: 'oscar') }
+  let(:fernanda) { Lita::User.create(157, mention_name: 'fernanda') }
+
+  def add_limit_order(order_id, user, type, created_at)
+    order = {
+      id: order_id,
+      user_id: user.id,
+      type: type,
+      created_at: created_at
+    }.to_json
+    subject.add_limit_order(order)
+    order
+  end
+
+  def setup_lunchers
+    [andres, fdom, oscar].each do |user|
+      lunch_assigner.add_to_lunchers(user.mention_name)
+      lunch_assigner.add_to_current_lunchers(user.mention_name)
+      lunch_assigner.add_to_winning_lunchers(user.mention_name)
+    end
+
+    lunch_assigner.add_to_lunchers(fernanda.mention_name)
+  end
+
+  describe '#add_limit_order' do
+    context 'first order added' do
+      before do
+        @order_id = SecureRandom.uuid
+        @order_time = Time.now
+        add_limit_order(@order_id, fdom, 'sell', @order_time)
+      end
+
+      it { expect(subject.orders.last).not_to be_nil }
+      it { expect(subject.orders.last['user_id']).to eq(fdom.id) }
+      it { expect(subject.orders.last['id']).to eq(@order_id) }
+      it { expect(subject.orders.last['type']).to eq('sell') }
+      it { expect(subject.orders.last['created_at']).to eq(@order_time.strftime('%F %T %z')) }
+    end
+
+    context 'with non empty orders' do
+      before do
+        add_limit_order(SecureRandom.uuid, andres, 'sell', Time.new(2018, 10, 2))
+        @order_id = SecureRandom.uuid
+        @order_time = Time.now
+        add_limit_order(@order_id, fdom, 'sell', @order_time)
```

Comment:
Aquí dejaría el primer `add_limit_order` dentro del bloque `before do`, ya que es una condición definida en el contexto "with non empty orders". Luego en cada `it` colocaría el segundo `(@order_id, fdom, 'sell', @order_time)`, ya que estás probando ese método cuando ya existe otra orden límite.